sdks/java/maven-archetypes/examples-java8/generate-sources.sh

```diff
@@ -22,7 +22,7 @@
 # Usage: Invoke with no arguments from any working directory.
 
 # The directory of this script. Assumes root of the maven-archetypes module.
-HERE="$(dirname $0)"
+HERE="$( cd "$( dirname '$0' )" && pwd )"
```

Comment:
I think you are right. `"$( cd "$( dirname '$0' )" && pwd )"` works because of `pwd`. I guess mvn change the directory when executing it. So when I run the script without mvn from other directory, there is an error. 

I tried with your suggestion. Actually no need backslash: `"$(dirname "$0")"` and works well for all cases.